app/components/pipeline/settings/main/modal/sync/component.js

```diff
@@ -59,7 +61,11 @@ export default class PipelineSettingsMainModalSyncComponent extends Component {
       .fetchFromApi('post', url)
       .then(() => {
         this.wasActionSuccessful = true;
-        this.args.closeModal();
+        if (this.args.syncType === 'pipeline') {
+          this.router.refresh();
+        } else {
+          this.args.closeModal();
+        }
```

Comment:
```suggestion
      .then(async () => {
        if (this.args.syncType === 'pipeline') {
          await this.shuttle
            .fetchFromApi(
              'get',
              `/pipelines/${this.pipelinePageState.getPipelineId()}`
            )
            .then(pipeline => {
              this.pipelinePageState.setPipeline(pipeline);
              this.pipelinePageState.forceReloadPipelineHeader();
            })
            .catch(() => {
              this.router.refresh();
            });
        }
        
        this.wasActionSuccessful = true;
        this.args.closeModal();
      })
```
If you follow this (this is what is done in the Sonar badge modal), you can reload the newly synchronized pipeline data without doing a hard refresh on the page.  The pipeline header will update automatically with the `this.pipelinePageState.forceReloadPipelineHeader()` helper (i.e., if the repository name changes then it will be updated after the sync completes).

I think it's a good idea to keep the `this.router.refresh()` if the API call fails for some reason.